File: wrappers/dotnet/indy-sdk-dotnet-test/WalletTests/GenerateWalletKeyTest.cs
using Hyperledger.Indy.WalletApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Newtonsoft.Json;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.WalletTests
{
    [TestClass]
    public class GenerateWalletKeyTest : IndyIntegrationTestBase
    {
        [TestMethod]
        public async Task TestGenerateWalletKeyWorks()
        {
            var key = await Wallet.GenerateWalletKeyAsync(null);
            var credentials = JsonConvert.SerializeObject(new { key = key, key_derivation_method = "RAW"});

            await Wallet.CreateWalletAsync(WALLET_CONFIG, credentials);
        }

        [TestMethod]
        public async Task TestGenerateWalletKeyWorksForSeed()
        {
            var config = JsonConvert.SerializeObject(new { seed = MY1_SEED });
            var key = await Wallet.GenerateWalletKeyAsync(config);

            Assert.AreEqual("CwMHrEQJnwvuE8q9zbR49jyYtVxVBHNTjCPEPk1aV3cP", key);

            var credentials = JsonConvert.SerializeObject(new { key = key, key_derivation_method = "RAW" });
            await Wallet.CreateWalletAsync(WALLET_CONFIG, credentials);
        }
    }
}

File: wrappers/android/src/androidTest/java/org/hyperledger/indy/sdk/wallet/GenerateWalletKeyTest.java
package org.hyperledger.indy.sdk.wallet;

import org.hyperledger.indy.sdk.IndyIntegrationTest;
import org.junit.Test;

import static org.junit.Assert.assertEquals;


public class GenerateWalletKeyTest extends IndyIntegrationTest {

	@Test
	public void testGenerateWalletKeyWorks() throws Exception {
		String key = Wallet.generateWalletKey(null).get();

		String credentials = "{ \"key\":\"" + key + "\", \"key_derivation_method\":\"RAW\"}";
		Wallet.createWallet(WALLET_CONFIG, credentials).get();
	}

	@Test
	public void testGenerateWalletKeyWorksForSeed() throws Exception {
		String config = "{ \"seed\":\"" + MY1_SEED + "\"}";
		String key = Wallet.generateWalletKey(config).get();
		assertEquals("CwMHrEQJnwvuE8q9zbR49jyYtVxVBHNTjCPEPk1aV3cP", key);

		String credentials = "{ \"key\":\"" + key + "\", \"key_derivation_method\":\"RAW\"}";
		Wallet.createWallet(WALLET_CONFIG, credentials).get();
	}
}